File: source/nestjs-petclinic-rest/src/repository/pet.repository.ts
import { Repository, EntityRepository } from 'typeorm';
import { CustomRepository } from './../typeorm/typeorm-ex.decorator';
import { Pet } from './../domain/entities/pet.entity';
import { IntentMerge } from './../intent/intent.decorators';

@IntentMerge()
@CustomRepository(Pet)
export class PetRepository extends Repository<Pet> {
}

File: source/spring-petclinic-rest/src/main/java/com/spring_petclinic/spring_petclinic_rest/data/PetRepository.java
package com.spring_petclinic.spring_petclinic_rest.data;

import org.springframework.data.jpa.repository.JpaRepository;
import com.spring_petclinic.spring_petclinic_rest.domain.models.Pet;
import com.spring_petclinic.spring_petclinic_rest.intent.IntentMerge;
import java.util.UUID;

/**
 * Spring Data JPA repository for the Pet entity.
 */
@IntentMerge
public interface PetRepository extends JpaRepository<Pet, Integer> {
}